sdk/core/Azure.Core.TestFramework/src/RunOnlyInTestModesAttribute.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace Azure.Core.TestFramework
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class | AttributeTargets.Assembly, AllowMultiple = false, Inherited = true)]
+    public class RunOnlyInTestModesAttribute : NUnitAttribute, IApplyToTest
+    {
+        public bool RecordOrPlayback { get; set; }
```

Comment:
Instead of separate properties, just pass the bitwise OR'd flags you want to support into the constructor, and might as well replace all the other similar attributes that would do the same thing with a specific case so we don't have duplication.